internal/app/carrier/repository.go

```diff
@@ -35,3 +41,42 @@ func (r *repository) InsertCarrierRequest(ctx context.Context, request model.Car
 	}
 	return nil
 }
+
+func (r *repository) UpdateCarrierRequest(ctx context.Context, parcel model.CarrierRequest, acceptStatus int, rejectStatus int, parcelStatus int, sourceTime time.Time) error {
+	//starting db transaction
+	tx, err := r.db.BeginTx(ctx, nil)
+	if err != nil {
+		log.Error().Err(err).Msg("[UpdateCarrierStatus] has failed to begin transactions.")
+		return fmt.Errorf("%v :%w", err, model.ErrTransaction)
```

Comment:
don't need Transaction error. it should return the error and handler should return internal server error